### PR TITLE
ui: wip

### PR DESCRIPTION
**Pull request by Domisys**

### CODE REVIEW

**Comment by Niktux** [comment]
ping @lebris : application done
- display and filtering
- pattern given by config
- introducing Process class

File: src/Services/XmlRPC/Server.php [hunk 1]
```diff
@@ -41,9 +44,21 @@ public function getProcessList()
...
+
+    public function extractApplicationList()
+    {
+        $apps = [];
+
+        foreach($this->getProcessList() as $process)
+        {
+            $apps[] = $process->getApplication();
+        }
+
```
Comment (Niktux) [inline src/Services/XmlRPC/Server.php:61]: missing array_unique

File: views/sidebar.twig [hunk 2]
```diff
@@ -24,6 +24,18 @@
         }]) %}
     {% endfor %}
     {{ ui_menu.multi('Servers', 'server', serverMenuItems) }}
-    {{ ui_menu.multi('Applications', 'desktop', serverMenuItems) }}
+
+	{% if apps is not empty %}
```
Comment (Niktux) [inline views/sidebar.twig]: cs

File: views/sidebar.twig [hunk 3]
```diff
@@ -24,6 +24,18 @@
...
+    {% set appMenuItems = [] %}
+    {% for app in apps %}
+        {% set appMenuItems = appMenuItems|merge([{
+            'route': 'applications',
+            'parameters': {'applicationName': app},
+            'text': app
+        }]) %}
+    {% endfor %}
+    {{ ui_menu.multi('Applications', 'desktop', appMenuItems) }}
+	{% endif %}
```
Comment (Niktux) [inline views/sidebar.twig]: cs

**Comment by Niktux** [comment]
add a condensed view for environments with many processes

view choice is persisted in html5 localStorage (compliant with refresh \o/)

TODO : use localStorage for filtering

File: src/Services/XmlRPC/Server.php (hunk 1 above)
Comment (Niktux) [inline src/Services/XmlRPC/Server.php:61]: Fix cdf369e07020dc625d88befb23f35124c1dc2988

File: views/partials/process_tile.twig [hunk 4]
```diff
@@ -19,13 +19,19 @@
...
+          <span class="processServer pull-right">
             <small>
                 <i class="fa fa-server"></i>
                 {{ process.server.name }}
             </small>
           </span>
 
+		  {% if process.application is not empty %}
+          <span class="processApplication">
+			<i class="fa fa-desktop"></i>
```
Comment (lebris) [inline views/partials/process_tile.twig]: There's some tabs in this file

File: assets/less/front.less [hunk 5]
```diff
@@ -30,3 +31,12 @@ a:hover, a:focus
 {
     text-decoration: none;
 }
+
+span.processServer,
+span.processApplication
+{
+	color: #cacaca;
+	i.fa {
+		margin-right: 5px;
```
Comment (lebris) [inline assets/less/front.less]: tabs

Comment (Niktux) [inline assets/less/front.less]: fix dd9c0720699f80237b5c8e5e14e0cdd3014c90cf

File: views/partials/process_tile.twig (hunk 4 above)
Comment (Niktux) [inline views/partials/process_tile.twig]: fix dd9c0720699f80237b5c8e5e14e0cdd3014c90cf

File: src/Services/XmlRPC/Process.php [hunk 6]
```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Supervisorg\Services\XmlRPC;
+
+use Puzzle\Configuration;
+class Process
```
Comment (lebris) [inline src/Services/XmlRPC/Process.php:7]: CS : missing empty line

File: src/Services/XmlRPC/CachedClient.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Supervisorg\Services\XmlRPC;
+
+class CachedClient implements Client
```
Comment (lebris) [inline src/Services/XmlRPC/CachedClient.php]: To move in the `Clients` directory

Comment (Niktux) [inline src/Services/XmlRPC/CachedClient.php]: fixed f09597f8838e4ab4c9fc6ad9c851441a88d4fe80

File: src/Services/XmlRPC/Process.php (hunk 6 above)
Comment (Niktux) [inline src/Services/XmlRPC/Process.php:7]: fixed b7fede66253cace628f8aecdf6a165825c6fb4e1

File: src/Controllers/Home/Controller.php [hunk 8]
```diff
@@ -17,41 +19,102 @@ class Controller
...
+
+    public function applicationsAction($applicationName)
+    {
+        $processes = [];
+        foreach($this->servers as $server)
+        {
+            $processes = array_merge($processes, $server->getProcessList());
+        }
+
+        $processes = new ApplicationFilterIterator(
```
Comment (lebris) [inline src/Controllers/Home/Controller.php:89]: TODO :  enabled/disabled in the configuration (and the view by server also maybe)

**Pull request merged**